### ISSUE TITLE
Support 2FA for users

### ISSUE THREAD

**Issue by hardillb** [FlowFuse/flowfuse#1880]
# Original Issue Description

To increase user security add support for 2FA as part of authentication
Possible options:
- TOTP (e.g. Google Authenticator or built in support in things like 1Password
- FIDO keys
- One time use backup codes

(Probably not SMS?)

## User Story
As a FlowFuse user,
I want the option to set up two-factor authentication (2FA),
So that my account, especially the admin account, is well-protected.

## Scope

1. Users can navigate to the security section within their account settings.
2. An option to set up TOTP is available and clearly labeled for users.
3. When users select TOTP, they are guided through the setup process, which includes scanning a QR code or entering a key into their authentication app.
4. Upon successful setup, a confirmation is displayed, indicating that TOTP 2FA is active for their account.
5. At login, after entering the correct account credentials, users are prompted to enter a one-time password generated by their authentication app.
6. Access to the account is granted only after the correct one-time password is provided, verifying the second factor.
7. PATs do not require 2FA

![Screenshot 2023-10-24 at 11 41 23](https://github.com/FlowFuse/flowfuse/assets/73583313/f5d8c5aa-f8e0-4d01-957a-550a76122ce4)

## Not in Scope for the first iteration

- FIDO keys
- Backup codes
- Trust this device X days
- Reminders to setup 2FA

**Comment by MarianRaphael** [comment]
Licence: Enterprise Tier

**Comment by ZJvandeWeg** [comment]
Possible package to use: https://github.com/beliven-it/fastify-totp

**Comment by MarianRaphael** [comment]
Docs update: https://github.com/FlowFuse/flowfuse/pull/3102

**Comment by knolleary** [comment]
I have raised #3106 as a follow up item as a next task under this topic. But the core requirement of this item has been delivered.
